cpp/src/join/distinct_hash_join.cu

```diff
@@ -337,6 +352,49 @@ distinct_hash_join<HasNested>::inner_join(rmm::cuda_stream_view stream,
 
   return {std::move(left_indices), std::move(right_indices)};
 }
+
+template <cudf::has_nested HasNested>
+std::pair<std::unique_ptr<rmm::device_uvector<size_type>>,
+          std::unique_ptr<rmm::device_uvector<size_type>>>
+distinct_hash_join<HasNested>::left_join(rmm::cuda_stream_view stream,
+                                         rmm::mr::device_memory_resource* mr) const
+{
+  cudf::thread_range range{"distinct_hash_join::left_join"};
+
+  size_type const build_table_num_rows{this->_build.num_rows()};
+  size_type const probe_table_num_rows{this->_probe.num_rows()};
+
+  // If output size is zero, return empty
+  if (probe_table_num_rows == 0) {
+    return {std::make_unique<rmm::device_uvector<size_type>>(0, stream, mr),
+            std::make_unique<rmm::device_uvector<size_type>>(0, stream, mr)};
+  }
+
+  auto build_indices =
+    std::make_unique<rmm::device_uvector<size_type>>(probe_table_num_rows, stream, mr);
+  auto probe_indices =
+    std::make_unique<rmm::device_uvector<size_type>>(probe_table_num_rows, stream, mr);
+  thrust::sequence(rmm::exec_policy_nosync(stream), probe_indices->begin(), probe_indices->end());
```

Comment:
Seems wasteful to manifest this sequence.  If we're willing to have the API guarantee that the left distinct join is going to emit the gather map in the same order as the left table, we could make this faster by:
- Not manifesting the trivial left table gather map.  Distinct left_join only returns the right table gather map.
- Not copying the left table columns when building the resulting table view of the join result, since we know the left table isn't getting rearranged in any way.  This part is the responsibility of the application and not libcudf, but we would take advantage of this.

Reply:
Goot point! Let me update this.